cmd/lncli/commands.go

```diff
@@ -1178,9 +1231,14 @@ func payInvoice(ctx *cli.Context) error {
 		return fmt.Errorf("pay_req argument missing")
 	}
 
+	amt = ctx.Int64("amt")
```

Comment:
I think there's a problem here if the user passes a payment request ID, in which case they generally will not pass an `amt` argument (so this will be 0). For this reason, you may have to do the percentage-fee-limit calculation on the backend, in `rpcserver.go`.